core/Scheduler/Scheduler.php

```diff
@@ -118,6 +125,14 @@ public function run()
                     continue;
                 }
 
+                $taskName = $task->getName();
+
+                if (!$this->acquireLockForTask($taskName, $task->getTTL())) {
+                    $this->logger->debug("Scheduler: '{task}' is currently executed by another process",
```

Comment:
Could there be other reasons why a lock could be not acquired? Just thinking about the message.. We are implying another run of the same task is already executed, but is it always the case? Should we make it perhaps more generic?

Reply:
The only other reason I could imagine is a database failure when writing the lock key. But in that case a database exception would be thrown, so this code shouldn't be executed anymore.

Reply:
The method to acquire the lock returns a boolean. True, if the lock was acquired - False, if the lock already exists. In case of any unexpected database error an exception will passed through. Can't see any reason why this should be handled differently 🤷